packages/eslint-plugin/src/tsconfig.json

```diff
@@ -3,6 +3,7 @@
     "compilerOptions": {
         "lib": ["es6", "dom"],
         "module": "commonjs",
+        "moduleResolution": "node16",
```

Comment:
this change was required to work with typescript-eslint v6, see https://github.com/typescript-eslint/typescript-eslint/issues/7284